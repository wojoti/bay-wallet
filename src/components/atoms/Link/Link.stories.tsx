import {ComponentMeta, ComponentStory} from '@storybook/react-native';
import {primary} from 'theme/colors';

import Link from './Link';

export default {
  title: 'Atoms/Link',
  component: Link,
  argTypes: {
    color: {
      control: {type: 'color'},
    },
    onPress: {action: 'onPress'},
    align: {
      control: {type: 'select'},
      options: ['left', 'right', 'auto', 'center', 'justify'],
    },
  },
} as ComponentMeta<typeof Link>;

const Template: ComponentStory<typeof Link> = args => <Link {...args} />;

export const Example = Template.bind({});
Example.args = {
  bold: false,
  color: primary,
  children: 'Type here for reset',
  align: 'left',
  testId: '',
};
