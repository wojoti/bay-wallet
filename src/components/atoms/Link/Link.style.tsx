import {StyleSheet} from 'react-native';

interface textProps {
  color?: string;
  bold?: boolean;
  align?: 'left' | 'auto' | 'right' | 'center' | 'justify' | undefined;
}
export const textStyle = ({color, bold, align}: textProps) =>
  StyleSheet.create({
    text: {
      margin: 0,
      padding: 0,
      color: color || 'black',
      fontWeight: bold ? 'bold' : '400',
      textAlign: align || 'left',
    },
  });

const style = StyleSheet.create({
  wrapper: {
    margin: 0,
    padding: 0,
    backgroundColor: 'none',
    borderColor: 'none',
  },
});
export default style;
