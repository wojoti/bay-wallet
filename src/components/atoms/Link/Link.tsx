import {PropsWithChildren} from 'react';
import {Pressable, Text} from 'react-native';
import style, {textStyle} from './Link.style';

export type LinkProps = PropsWithChildren<{
  testId?: string;
  bold?: boolean;
  color?: string;
  align?: 'left' | 'auto' | 'right' | 'center' | 'justify';
  onPress: () => void;
}>;

function Link({
  children,
  onPress,
  testId,
  color,
  bold = false,
  align,
}: LinkProps) {
  return (
    <Pressable
      accessibilityRole="button"
      style={style.wrapper}
      testID={testId}
      onPress={onPress}>
      <Text style={textStyle({color, bold, align}).text}>{children}</Text>
    </Pressable>
  );
}
export default Link;
