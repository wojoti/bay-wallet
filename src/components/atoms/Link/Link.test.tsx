import '@testing-library/jest-native/extend-expect';
import {fireEvent, render, screen} from '@testing-library/react-native';
import TestRenderer from 'react-test-renderer';
import Link, {LinkProps} from './Link';

const handlePress = jest.fn();
const props: LinkProps = {
  children: 'test link',
  onPress: handlePress,
  testId: 'test-link-id',
};

const modifiedProps: LinkProps = {
  ...props,
  bold: true,
};

test('should match snapshot', () => {
  const tree = TestRenderer.create(<Link {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render matching link with text', () => {
  render(<Link {...props} />);
  const linkElement = screen.getByTestId('test-link-id');
  expect(linkElement).toBeOnTheScreen();
  expect(linkElement).toHaveTextContent('test link');
});

test('should handle onPress event', () => {
  render(<Link {...props} />);
  const linkElement = screen.getByTestId('test-link-id');
  fireEvent.press(linkElement);
  expect(handlePress).toHaveBeenCalledTimes(1);
});

test('should render link with modified style through props', () => {
  render(<Link {...modifiedProps} />);
  const linkElement = screen.getByTestId('test-link-id');
  expect(linkElement).toBeOnTheScreen();
  const textElement = screen.getByText('test link');
  expect(textElement).toBeOnTheScreen();
  expect(textElement).toHaveStyle({textDecorationLine: 'underline'});
});
