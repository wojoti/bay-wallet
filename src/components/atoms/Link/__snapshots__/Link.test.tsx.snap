// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "none",
      "borderColor": "none",
      "margin": 0,
      "padding": 0,
    }
  }
  testID="test-link-id"
>
  <Text
    style={
      {
        "color": "black",
        "fontWeight": "400",
        "margin": 0,
        "padding": 0,
        "textAlign": "left",
      }
    }
  >
    test link
  </Text>
</View>
`;
